pkg/controller/common/stackmon/monitoring/monitoring.go

```diff
@@ -18,10 +18,38 @@ type HasMonitoring interface {
 	MonitoringAssociation(ref commonv1.ObjectSelector) commonv1.Association
 }
 
+// IsReconcilable return true if a resource has at least one association defined in its specification
+// and all defined associations are configured.
+func IsReconcilable(resource HasMonitoring) bool {
+	return IsDefined(resource) && isMetricsAssocConfigured(resource) && isLogsAssocConfigured(resource)
+}
+
+// IsDefined return true if a resource has at least one association for Stack Monitoring defined in its specification
+// (can be one for logs or one for metrics).
 func IsDefined(resource HasMonitoring) bool {
 	return IsMetricsDefined(resource) || IsLogsDefined(resource)
 }
 
+func isMetricsAssocConfigured(resource HasMonitoring) bool {
```

Comment:
I guess it is a little bit confusing that this check passes when there is no association configured at all! Not sure if it would help to either add a comment or to add an `Optional` somewhere in the function name. Or maybe something like
`isMetricsAssocConfiguredIfDefined` and
`isLogsAssocConfiguredIfDefined`

Reply:
I totally understand. With `IsReconcilable` these two methods are only used once, so I think we can just remove them and inline the code in `IsReconcilable`.